### DISCUSSION TITLE
怎么提取图片的特征值呢

### DISCUSSION THREAD

**Discussion by holmofy** [milvus-io/milvus#16282] [Q&A and General discussion] [answered]
怎么提取图片的特征值呢。目前基于java语言，用的是pHash算法。milvus有提供向量化功能吗，java-sdk可以直接传图片吗？

**Comment by yhmo** [comment, marked as answer]
Milvus has no ability to convert pictures into embeddings. But there is another open-source solution to help you, it is https://github.com/towhee-io/towhee

This python tool can convert picture/voice/words into embeddings, by just several lines:
```
from towhee import pipeline
embedding_pipeline = pipeline('image-embedding')
embedding = embedding_pipeline('https://docs.towhee.io/img/logo.png')
```

So, you can use towhee+milvus to quickly build a service, including embedding generating and embedding searching.
